### PR TITLE
[Mailer] Deprecate OhMySmtp Transport, Create MailPace transport

### LINKED ISSUE #46340: [Mailer] Create a new Mailpace Bridge to replace OhMySMTP Bridge
### Description

See at https://docs.mailpace.com/guide/moving_from_ohmysmtp/

OhMySMTP has been renamed to Mailpace, we should create a new bridge to support to new API url and header names, and deprecate the OhMySMTP one.

### Example

_No response_

### PR DESCRIPTION
**Pull request by Holicz**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | yes
| Tickets       | Fix #46340
| License       | MIT
| Doc PR        | symfony/symfony-docs#16896

OhMySMTP duplicate was created and renamed to MailPace. OhMySMTP was deprecated.

### CODE REVIEW

File: src/Symfony/Component/Mailer/Bridge/OhMySmtp/Transport/OhMySmtpApiTransport.php [hunk 1]
```diff
@@ -24,8 +24,12 @@
 use Symfony\Contracts\HttpClient\HttpClientInterface;
 use Symfony\Contracts\HttpClient\ResponseInterface;
 
+trigger_deprecation('symfony/oh-my-smtp-mailer', '6.2', 'The "%s" class is deprecated.', OhMySmtpApiTransport::class);
```
Comment (welcoMattic) [inline src/Symfony/Component/Mailer/Bridge/OhMySmtp/Transport/OhMySmtpApiTransport.php]: Please mention the new Bridge as replacement here: "…deprecated, use Symfony\Component\Mailer\Bridge\MailPace\Transport\MailPaceApiTransport instead."

File: src/Symfony/Component/Mailer/Bridge/OhMySmtp/Transport/OhMySmtpSmtpTransport.php [hunk 2]
```diff
@@ -20,8 +20,12 @@
 use Symfony\Component\Mime\Message;
 use Symfony\Component\Mime\RawMessage;
 
+trigger_deprecation('symfony/oh-my-smtp-mailer', '6.2', 'The "%s" class is deprecated.', OhMySmtpSmtpTransport::class);
```
Comment (welcoMattic) [inline src/Symfony/Component/Mailer/Bridge/OhMySmtp/Transport/OhMySmtpSmtpTransport.php]: Please mention the new Bridge as replacement here: "…deprecated, use Symfony\Component\Mailer\Bridge\MailPace\Transport\MailPaceSmtpTransport instead."

**Review by welcoMattic — COMMENTED** [review]
Looks good, thank you @Holicz. I have added some notes

File: src/Symfony/Component/Mailer/Bridge/MailPace/Transport/MailPaceApiTransport.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                'User-Agent' => 'MailPace Symfony Mailer',
+            ],
+            'json' => $this->getPayload($email, $envelope),
+        ]);
+
+        try {
+            $statusCode = $response->getStatusCode();
+            $result = $response->toArray(false);
+        } catch (DecodingExceptionInterface) {
+            throw new HttpTransportException('Unable to send an email: '.$response->getContent(false).sprintf(' (code %d).', $statusCode), $response);
```
Comment (stof) [inline src/Symfony/Component/Mailer/Bridge/MailPace/Transport/MailPaceApiTransport.php]: IMO, we should attach the caught exception as previous exception.

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by Holicz** [comment]
All comments were taken into account.

PS: What is the process for this? Should I resolve the comments when I push the changes or should the original author of the comment review the touched code once again and resolve it himself?

Thanks!

**Comment by welcoMattic** [comment]
@Holicz you can resolve comment (if needed, we can unresolve it)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Holicz.

**Comment by fabpot** [comment]
@Holicz Can you submit a PR on symfony/recipes and one for the docs?

**Comment by Holicz** [comment]
Sure,

Docs: https://github.com/symfony/symfony-docs/pull/16896
Recipes: https://github.com/symfony/recipes/pull/1104

**Pull request closed without merging**